onadata/apps/logger/management/commands/update_attachment_storage_bytes.py

```diff
@@ -0,0 +1,177 @@
+
+        # Dictionary which contains the total of bytes for each user. Useful
+        # to avoid another aggregate query on attachments to calculate users
+        # global usage.
+        total_per_user = defaultdict(int)
+
+        # Release any locks on the users' profile from getting submissions
+        UserProfile.objects.exclude(
+            metadata__attachments_counting_status='complete'
+        ).update(
+            metadata=ReplaceValues(
+                'metadata',
+                updates={'submissions_suspended': False},
+            ),
+        )
+
+        # Get only xforms whose users' storage counters have not been updated yet.
+        up_queryset = UserProfile.objects.values_list('user_id', flat=True).filter(
+            metadata__attachments_counting_status='complete'
+        )
+        xforms = (
+            XForm.objects.exclude(user_id__in=up_queryset)
+            .values('pk', 'user_id', 'user__username')
+            .order_by('user_id')
+        )
+
+        last_xform = None
+        for xform in xforms:
+
+            if not last_xform or (last_xform['user_id'] != xform['user_id']):
+                # Retrieve or create user's profile.
+                (
+                    user_profile,
+                    created,
+                ) = UserProfile.objects.get_or_create(user_id=xform['user_id'])
+
+                # Some old profiles don't have metadata
+                if user_profile.metadata is None:
+                    user_profile.metadata = {}
+
+                # Set the flag to true if it was never set.
+                if not user_profile.metadata.get('submissions_suspended'):
+                    # We are using the flag `submissions_suspended` to prevent
+                    # new submissions from coming in while the
+                    # `attachment_storage_bytes` is being calculated.
+                    user_profile.metadata['submissions_suspended'] = True
+                    user_profile.save(update_fields=['metadata'])
+
+                # if `last_xform` is not none, it means that the `user_id` is
+                # different from the previous one in the loop so the user
+                # profile must be updated
+                if last_xform:
+                    self.update_user_profile(
+                        last_xform, total_per_user[last_xform['user_id']]
+                    )
+
+            # write out xform progress
+            if self.verbosity >= 1:
+                self.stdout.write(
+                    f"Calculating attachments for xform_id #{xform['pk']}"
+                    f" (user {xform['user__username']})"
+                )
+            # aggregate total media file size for all media per xform
+            # We cannot get the sum of all attachments for on xform because
+            # we need to make two joins with Instance and XForm models.
+            # It is really slow on big databases. So, to avoid that, we make
```

Comment:
After https://github.com/kobotoolbox/kobocat/pull/828#discussion_r896194353, I thought this ranged from 10 seconds for 22k attachments to 31 minutes for 4m attachments, which we thought was acceptable for this `submissions_suspended` approach. Does this technique perform better? Also, fwiw, there ought to be one join, not two (query from the same link above, just formatted):
```sql
SELECT
  sum(logger_attachment.media_file_size),
  count(*)
FROM
  "logger_attachment"
  INNER JOIN "logger_instance" ON (
    "logger_attachment"."instance_id" = "logger_instance"."id"
  )
WHERE
  "logger_instance"."xform_id" in (
    select
      id
    from
      logger_xform
    where
      user_id = 13632
  );
```

Reply:
We did test it with same set of data and it was way faster. I don't know if it because of the load at the moment or the upgrade of PostgreSQL, but there are no gains now from my today tests (prod and local). 
I'm gonna revert all this block because it brings way more complexity with no benefits after all. 
If you want, I can rebase the branch after your (final) review to avoid making GitHub confused.

Reply:
Thanks :+1: I like the new approach (I don't know whether a rebase is needed, but I'll leave the decision to you)

Reply:
Keep as-is.